zebra-state/tests/basic.rs

```diff
@@ -56,15 +56,15 @@ static GET_TIP_TRANSCRIPT_MAINNET: Lazy<Vec<(Request, Result<Response, TransErro
         let hash0 = block0.as_ref().into();
         let hash1 = block1.as_ref().into();
         vec![
-            // Insert higher block first, lower block second
-            (
-                Request::AddBlock { block: block1 },
-                Ok(Response::Added { hash: hash1 }),
-            ),
+            // Insert the blocks in order
```

Comment:
do we have an out of order test?

Reply:
We can't enable it by default until #1011 is merged, but you're right, we should have an out of order test.

(Before this PR, all the tests were out of order.)

Reply:
The CheckpointVerifier `continuous_blockchain_test` is unordered, which is why I disabled it:
https://github.com/ZcashFoundation/zebra/pull/1012/files#diff-4b3b16117479af65b38872f84005c2e1R213

But we should have tests that are guaranteed out of order, rather than just random execution order.

Reply:
Added those tests in 2736ec7, then commented out the failing out-of-order tests in 533a401.

I also squashed the fixes into the branch and force-pushed, so we can merge this branch using a rebase. That way, we can revert 533a401 once we fix #1011.

3a51d73 is the squashed commit, the others are the same.

The old branch is here:
https://github.com/teor2345/zebra/tree/state-order-2020-09-07

Reply:
thank you for the old branch ref, for anyone else who needs the incremental heres the link to the compare

https://github.com/teor2345/zebra/compare/state-order-2020-09-07...teor2345:state-order